### DISCUSSION TITLE
Can handle eleventy German umlauts?

### DISCUSSION THREAD

**Discussion by dancesWithCycles** [11ty/eleventy#1998] [Q&A] [answered]
Hi folks, Thank you so much for providing and maintaining this repository. Please accept my appreciation!

I am new to eleventy and just observed that it is not handling German umlauts in template files like ```index.liquid``` as I expected. Here is the snippet.

```
----
title: Software Ingenieur Begerad
layout: main
---

<h2>Software Ingenieur Begerad</h2>

<p>Willkommen auf der Webseite des Unternehmens <b>{{ site.organization.name }}</b>. Es bietet Ihnen Ingenieursdienstleistungen (Beratung, Entwurf, Planung, Konstruktion, Ausführung, Betrieb, Vertrieb und Überwachung) für <b>Softwarefreiheit</b> an.</p>
...
```

And so on... BTW, the German umlauts are encoded in the correct manner here in the GitHub discussion. But it looks bad on my [website](https://www.swingb.de).

Am I doing something wrong? Is it better to put content like this always in dedicated page files with the appropriate markup language? If the answer is yes, do you know an example that shows how to do it for the index/front page?

I appreciate and hint the steers me in the right direction.

Cheers!

**Comment by cfjedimaster** [comment]
I'm not able to recreate this. Do you have your code up on a repo where I can grab it and run it myself?

**Comment by dancesWithCycles** [comment]
Hi folks, Please let me add another observation. When I run this website locally with ```npx @11ty/eleventy --serve``` everything is fine. The problem appears when I copy all files onto my webserver. Is copying wrong? Is there a deployment procedure for static sites generated with eleventy?

**Reply by cfjedimaster** [reply]
I'd look at your deployment. If the files are getting munged when you copy them, how are you doing the copy? To your general question of "is there a deployment procedure", there isn't one *specific* to Eleventy or the Jamstack as a whole. There *are* multiple deployment options out there that cater to the Jamstack though. So for example, I use Netlify for my hosting, and they can connect to my GitHub repository so that when I commit  a change, they get notified, get my files, run Eleventy and then deploy the files.

**Comment by dancesWithCycles** [comment]
Here is the [code](git@github.com:Software-Ingenieur-Begerad/website.git). Please give me some seconds to push the latest version.

git@github.com:Software-Ingenieur-Begerad/website.git

**Comment by dancesWithCycles** [comment]
Ok, latest version is available in the main branch.

git@github.com:Software-Ingenieur-Begerad/website.git

**Comment by cfjedimaster** [comment]
For other folks, that URL is: https://github.com/Software-Ingenieur-Begerad/website

Yeah, it seems fine to me. How are you deploying?

**Reply by dancesWithCycles** [reply]
I am copying the content of  the **_site** directory into the **DocumentRoot** directory of the web server.

**Comment by dancesWithCycles** [comment]
I found the following in the dev tool of my browser.

```
The character encoding of the HTML document was not declared. The document will render with garbled text in some browser configurations if the document contains characters from outside the US-ASCII range. The character encoding of the page must be declared in the document or in the transfer protocol. www.swingb.de
```

**Comment by dancesWithCycles** [comment]
My interpretation of the dev tool browser message is that I should add character encoding in the **main layout**.

```
---
title: Default title
---
<html>
<head>
<title>{{ title }}</title>
{% comment %} add link tag to the layout {% endcomment %}
<link rel="stylesheet" href="/style/app.css">
</head>
<body>
    <div class="header">
    <a href="/">Home</a> ~ <a href="/archives">Archives</a>
    </div>
    <div class="content">
    {{ content }}
    </div>

    <footer>
	<p>
	Created by {{ site.author.name }}, follow them on <a href="https://twitter.com/{{site.author.twitter}}">Twitter</a>.
	</p>
</footer>
</body>
</html>
```

This is the current version without character encoding.

**Comment by cfjedimaster** [comment]
So perhaps:

```
<html lang="en">
```

and also a 

```
<meta charset="utf-8">
```

Kinda guessing here. :) It's weird that a simple file copy is breaking it for you.

**Comment by dancesWithCycles** [comment, marked as answer]
```
---
title: Default title
---
<html>
<head>
<meta charset="utf-8">
<title>{{ title }}</title>
{% comment %} add link tag to the layout {% endcomment %}
<link rel="stylesheet" href="/style/app.css">
</head>
<body>
    <div class="header">
    <a href="/">Start</a> &bull; <a href="/contact">Kontakt</a>
    </div>
    <div class="content">
    {{ content }}
    </div>

    <footer>
	<p>
	{{ site.author.name }} auf <a target="_blank" href="https://github.com/{{site.author.github}}">GitHub</a> folgen.
	</p>
</footer>
</body>
</html>
```

Do you see the difference?

It really was just the **charset** tag missing.

```
<meta charset="utf-8">
```

Thank you so much @cfjedimaster for walking me there.

Cheers!

**Reply by cfjedimaster** [reply]
Glad to help - it's in my default HTML template and I honestly don't even think about it.
